Expensify/App
# Reroute user trying to link to VBA steps they cannot complete

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/168078

### Tests / QA Steps (web/mWeb only)
1. Sign up with a new account.
1. Create a new workspace.
1. Click on `Settings` -> `Your Workspace` -> `Connect bank account`.
1. Attempt to go to the following routes: 
    - `/bank-account/company`
    - `/bank-account/personal-information`
    - `/bank-account/contract`
    - `/bank-account/validate`
    - `/bank-account/enable`

    And verify that for each you remain on this page (note that the URL will change but the UI should not:
    
    ![image](https://user-images.githubusercontent.com/47436092/138982290-81d4a7f2-94ef-4b9b-b7f8-f2a6205b71b5.png)

1. Hit `Enter Manually`
1. Enter `011401533` for the routing number and `1111222233331111` for the account number, check the box, then hit save and continue.
1. Attempt to go to the following routes:

    - `/bank-account/personal-information`
    - `/bank-account/contract`
    - `/bank-account/validate`
    - `/bank-account/enable`
    
    And verify that for each you remain on the company step.

1. Hit the back arrow.
1. Verify that you see the account and routing number you entered.
1. Attempt to go to `bank-account/personal-information` and verify that you are taken back to the company step.
1. Enter company details, then press save and continue.
1. Attempt to go to the following routes:
    - `/bank-account/contract`
    - `/bank-account/validate`
    - `/bank-account/enable`
    
    And verify that for each you remain on the requestor (personal-information) step.

1. Hit the back arrow.
1. Verify that you see the company details you entered before.
1. Attempt to go to `bank-account/contract` and verify that you are taken back to the requestor step.
1. Enter personal details, then press save and continue. (TODO: I am not able to get past this, even on `main`)

### Tested On
